File: php/features.php
<?php
declare(strict_types=1);

// DeckOfCards SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class DeckOfCardsFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new DeckOfCardsBaseFeature();
            case "test":
                return new DeckOfCardsTestFeature();
            default:
                return new DeckOfCardsBaseFeature();
        }
    }
}

File: py/features.py
# DeckOfCards SDK feature factory

from feature.base_feature import DeckOfCardsBaseFeature
from feature.test_feature import DeckOfCardsTestFeature


def _make_feature(name):
    features = {
        "base": lambda: DeckOfCardsBaseFeature(),
        "test": lambda: DeckOfCardsTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
